eng/scripts/Create-Apiview-Token-TypeSpec.ps1

```diff
@@ -0,0 +1,110 @@
+[CmdletBinding()]
+param (
+    [ValidateNotNullOrEmpty()]
+    [string] $Reviews,
+    [ValidateNotNullOrEmpty()]
+    [string] $WorkingDir,
+    [ValidateNotNullOrEmpty()]
+    [string] $OutputDir,
+    [string] $GitPat = ""
+)
+
+Set-StrictMode -Version 3
+$sparseCheckoutFile = ".git/info/sparse-checkout"
+$globalPackageJson = "package.json"
+
+
+function Sparse-Checkout($branchName, $packagePath)
+{
+    if (Test-Path $sparseCheckoutFile)
+    {
+        Remove-Item $sparseCheckoutFile -Force
+    }    
+    git sparse-checkout init --no-cone
+    git sparse-checkout set $packagePath $globalPackageJson

```

Comment:
Is this actually needed? Doesn't sparse-checkout automatically include the files in the root?

Reply:
I will check this again.

Reply:
1. Sparse checkout by default WILL get files in the root. Also if you're not doing negative globs, I recommend not doing `--no-cone` as it's not necessary.
2. I think it would be better to put the `git clone` command into this function.
3. You don't need the conditional sparse checkout removal. `init` is idempotent and `set` will overwrite any previous contents of the sparse config file.

Reply:
Thanks @weshaggard  and @benbp. This pipeline will process API review gen for multiple reviews when we add support for review upgrade and it's possible to have each review generated from different forks of specs repo or different branches within same repo. `git clone` is skipped when iterating over the reviews if it's same repo. I have removed `--no-clone` and sparse checkout config file removal.